src/biotite/sequence/profile.py

```diff
@@ -0,0 +1,165 @@
+    position_gaps : ndarray, dtype=int, shape=n
+        Array which indicates the number of gaps at each position.
+    alphabet : Alphabet, length=k
+        Alphabet of sequences of sequence profile
+    """
+
+    def __init__(self, position_frequency_matrix, position_gaps, alphabet):
+        self.position_frequency_matrix = position_frequency_matrix
+        self.position_gaps = position_gaps
+        self.alphabet = alphabet
+
+    def __repr__(self):
+        """Represent SequenceProfile as a string for debugging."""
+        return f"SequenceProfile(np.{np.array_repr(self.position_frequency_matrix)}, " \
+               f"np.{np.array_repr(self.position_gaps)}, Alphabet({self.alphabet}))"
+
+    def __eq__(self, item):
+        if not isinstance(item, SequenceProfile):
+            return False
+        if not np.array_equal(self.position_frequency_matrix, item.position_frequency_matrix):
+            return False
+        if not np.array_equal(self.position_gaps, item.position_gaps):
+            return False
+        if not self.alphabet == item.alphabet:
+            return False
+        return True
+
+    @staticmethod
+    def from_alignment(alignment):
+        """
+        Get an object of :class:`SequenceProfile` from an object of :class:`Alignment`
+        """
+        sequences = alignment.get_gapped_sequences()
+        alphabet = alignment.sequences[0].get_alphabet().get_symbols()
+        position_frequency_matrix = np.zeros((len(sequences[0]), len(alphabet)), dtype=int)
+        position_gaps = np.zeros(len(sequences[0]), dtype=int)
+        for i in range(len(sequences[0])):
+            for j in range(len(sequences)):
+                if sequences[j][i] != "-":
+                    position_frequency_matrix[i, alphabet.index(sequences[j][i])] += 1
+                else:
+                    position_gaps[i] += 1
+        return SequenceProfile(position_frequency_matrix, position_gaps, alignment.sequences[0].get_alphabet())
+
+    def to_consensus(self):
+        """
+        Get the consensus sequence for :class:`SequenceProfile`
+
+        Returns
+        -------
+        consensus: str
+            The calculated consensus sequence
+        """
+        # https://en.wikipedia.org/wiki/International_Union_of_Pure_and_Applied_Chemistry#Amino_acid_and_nucleotide_base_codes
+        if set(self.alphabet.get_symbols()) == set(_NUC_DNA_ALPH.get_symbols()):
+            return self._dna_to_consensus()
+        elif set(self.alphabet.get_symbols()) == set(_NUC_RNA_ALPH.get_symbols()):
+            return self._rna_to_consensus()
+        elif set(self.alphabet.get_symbols()) == set(_PROT_ALPH.get_symbols()):
+            return self._prot_to_consensus()
+        return self._custom_to_consensus()
+
+    def _codes_to_iupac(self, frequency, codes):
+        if np.sum(frequency) == 0:
+            return "-"
+        position_max = np.where(frequency == np.amax(frequency))[0].tolist()
```

Comment:
The current implementation runs this for every position in the profile. However `np.where()` and `np.amax()` are also able to handle multidimensional arrays. So maybe it is possible to run this once for the entire profile to increase the efficiency.

Reply:
Hi @padix-key , I chose this implementation because I need to get multiple maximum values for each row of ndarray symbols. I did not find a solution to to this once for a multidimensional array. Do you have an idea how to do this?

Reply:
For `np.where()` you are probably right, I also have currently no direct solution to solve this in a single function call. However, you could at least obtain the maximum value in each column with `np.max(self._symbols, axis=1)`